pints/tests/test_toy_stochastic_degradation_model.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+#
+# Tests if the stochastic degradation (toy) model works.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import unittest
+import numpy as np
+import pints
+import pints.toy
+
+
+class TestStochasticDegradation(unittest.TestCase):
+    """
+    Tests if the stochastic degradation (toy) model works.
+    """
+
+    def test_start_with_zero(self):
+        # Test the special case where the initial concentration is zero
+        from pints.toy import StochasticDegradationModel
+        model = StochasticDegradationModel(0)
+        times = [0, 1, 2, 100, 1000]
+        parameters = [0.1]
+        values = model.simulate(parameters, times)
+        self.assertEqual(len(values), len(times))
+        self.assertTrue(np.all(values == np.zeros(5)))
+
+    def test_start_with_twenty(self):
+        # Run small simulation
+        model = pints.toy.StochasticDegradationModel(20)
+        times = [0, 1, 2, 100, 1000]
+        parameters = [0.1]
+        values = model.simulate(parameters, times)
+        self.assertEqual(len(values), len(times))
+        self.assertEqual(values[0], 20)
+        self.assertEqual(values[-1], 0)
+
+    def test_suggested(self):
+        model = pints.toy.StochasticDegradationModel(20)
+        times = model.suggested_times()
+        parameters = model.suggested_parameters()
+        self.assertTrue(np.all(times == np.linspace(0, 100, 101)))
+        self.assertEqual(parameters, 0.1)
+        values = model.simulate(parameters, times)
+        self.assertEqual(values[0], 20)
+        self.assertEqual(values[-1], 0)
+
+    def test_simulate(self):
+        model = pints.toy.StochasticDegradationModel(20)
+        parameters = [0.1]
+        times = np.linspace(0, 100, 101)
+        values = model.simulate(parameters, times)
+
+        # Test output of Gillespie algorithm
```

Comment:
These tests need to go!
A unit test should test that a class does what its supposed to do, *but not how it actually does it*!
So you never check internals / private properties of a class, you only check that the methods return what they ought to return. (It gets a bit trickier if the methods update some internal state of the object, but that's not the case here!)
(And don't solve this by adding a method to return these properties either ;-) )

So all your tests should be tests of the `values` returned

Reply:
Done

Reply:
What would happen to `values` if it went wrong? Can we check for that?

Reply:
Not without knowing the internals, no.

Reply:
Actually I've figured a work-round. Give me a few mins

Reply:
Actually, no I haven't -- can we keep this as is please?

Reply:
Why don't you seed the random generator, print the output, check it manually, and then hardcode those values in the test?

Reply:
I've created separate functions that do the interpolation and return the raw values (and added tests for these). Let us know what you think @MichaelClerx

Reply:
I still don't quite understand why that's needed. Will users benefit from having these methods as part of the API? If so, then great, if not then this is not the way to go.
But if we can't tell whether its working from the output of simulate(), then why do we care?

(I found this article https://enterprisecraftsmanship.com/2017/10/23/unit-testing-private-methods/ that says the stuff I'm saying a bit better, even if it doesn't give that many arguments)